### PR TITLE
Update workflow_utils build modules and remove ncio module hack

### LINKED ISSUE #407: Update ncio module to hpc-stack version and remove hack for workflow_utils build
Opened an issue with hpc-stack to get the ncio/1.0.0 module added to the stack on Jet: https://github.com/NOAA-EMC/hpc-stack/issues/296

The ncio/1.0.0 module is already available in the stack on WCOSS-Dell, Hera, and Orion.

Remove ncio checkout/hack and add to modulefiles (if missing) once it's available in the stack on Jet.

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
This PR removes the ncio module hack in the workflow_utils build and updates the workflow_utils build modules on all supported platforms to add the ncio module available via hpc-stack.

Close #407

### CODE REVIEW

**Comment by KateFriedman-NOAA** [comment]
Completing 2.5 cycle C192C96L127 test on WCOSS-Dell with final PR branch before submitting this after approval.

File: modulefiles/workflow_utils.hera [hunk 1]
```diff
@@ -20,10 +20,9 @@ module load sp/2.3.3
 module load ip/3.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
-#module load ncio/1.0.0
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
Comment (aerorahul) [inline modulefiles/workflow_utils.hera]: please move this right after `png`.
One of the `w3` libraries used to have netcdf dependency (or may be still does).

File: modulefiles/workflow_utils.jet [hunk 2]
```diff
@@ -24,9 +24,9 @@ module load sp/2.3.3
 module load ip/3.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
Comment (aerorahul) [inline modulefiles/workflow_utils.jet]: same comment as above.

File: modulefiles/workflow_utils.orion [hunk 3]
```diff
@@ -20,10 +20,9 @@ module load sp/2.3.3
 module load ip/3.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
-#module load ncio/1.0.0
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
Comment (aerorahul) [inline modulefiles/workflow_utils.orion]: same here.

File: modulefiles/workflow_utils.wcoss_dell_p3 [hunk 4]
```diff
@@ -20,10 +20,9 @@ module load sp/2.3.3
 module load ip/3.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
-#module load ncio/1.0.0
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
Comment (aerorahul) [inline modulefiles/workflow_utils.wcoss_dell_p3]: and here.

**Review by aerorahul — CHANGES_REQUESTED** [review]
Looks good.
One small request for change.

File: modulefiles/workflow_utils.hera (hunk 1 above)
Comment (KateFriedman-NOAA) [reply modulefiles/workflow_utils.hera]: Done

File: modulefiles/workflow_utils.jet (hunk 2 above)
Comment (KateFriedman-NOAA) [reply modulefiles/workflow_utils.jet]: Done

File: modulefiles/workflow_utils.orion (hunk 3 above)
Comment (KateFriedman-NOAA) [reply modulefiles/workflow_utils.orion]: Done

File: modulefiles/workflow_utils.wcoss_dell_p3 (hunk 4 above)
Comment (KateFriedman-NOAA) [reply modulefiles/workflow_utils.wcoss_dell_p3]: Done

**Comment by KateFriedman-NOAA** [comment]
Moved hdf5 and netcdf module loads earlier in workflow_utils modulefiles since they are prereqs for some other modules loaded later in modulefiles. Retested workflow_utils build on WCOSS-Dell, Hera, Jet, and Orion - successful.

**Review by aerorahul — APPROVED** [review]
looks good.
thanks for removing hacks.

**Pull request merged**